cmake/emu/qemu.cmake

```diff
@@ -224,6 +224,17 @@ if(CONFIG_X86_IAMCU)
     )
 endif()
 
+if(CONFIG_X86_LONGMODE)
+  # QEMU doesn't like 64-bit ELF files. Since we don't use any >4GB
```

Comment:
Huh, interesting. I wonder if upstream is fixing this.

Reply:
I don't think they plan to.  I think the concern is that -- in the absence of a more developed firmware environment like UEFI -- simply transferring control to a 64-bit entry point doesn't make sense. I think they're trying to avoid someone slapping together a 64-bit ELF binary and expecting it to work (and then complaining about it).